### ISSUE TITLE
Handling strings and objects differently in message (wrt AppInsights)

### ISSUE THREAD

**Issue by pro-panda** [NLog/NLog#4103]
This is a very specific scenario where I can't find a way to configure NLog - 
I'm having two targets - File and Application Insights. Most of my logs are strings, but some of them are objects (nested key value pairs of strings). In Application Insights, I want to keep my Raw data (entire message payload) as json parseable - hence the JsonLayout.

**Current Approach:**
To achieve this, I send `myObject` as a separate property while skipping encoding - which allows the object to not get converted to a string and get parsed in the RawData.
```  
<targets>
    <target name="logfile" xsi:type="File" layout="${event-properties:item=myObject}${message}" />
    <target xsi:type="ApplicationInsightsTarget" name="aiTarget">
      <layout xsi:type="JsonLayout">
        <attribute name="MyItem" layout="${mdlc:item=MyItem:whenEmpty=Default}"/>
        <attribute name="Message" layout="${message}"/>
        <attribute name="myObject" layout="${event-properties:item=myObject}" encode='false' />
      </layout>
    </target>
</targets>
```

```
{
    "MyItem": "MyItemValue",
    //"Message": <message key is absent since I pass empty string while logging the object>
    "myObject": {
        "key1": "value1"
        ....
    }
}
```

**Failred Approaches:**
If remove `encoding="false"` or serialize the string myself, it loses formating and the result looks like (notice the outermost quotes on the value of myObject)- 
```
{
    "MyItem": "MyItemValue",
    "myObject": "{ \"key1\": \"value1\"....}"
}
```

If I skip encoding and send the object in the message field, then my string messages are missing quotes in the Raw payload, leaving that JSON un-parseable,
```
{
    "MyItem": "MyItemValue",
    "Message": Some message - notice the missing quotes on this value
}
```
**Issue:**
While my original approach works, all event properties are also logged to CustomDimensions of Application Insights (see [L105](https://github.com/microsoft/ApplicationInsights-dotnet-logging/blob/810560b1e6c302555c85b84d5512d60d83be7f83/LOGGING/src/NLogTarget/ApplicationInsightsTarget.cs#L105)), which causes duplicacy of data.

To summarize,
1. Is there any way I can log both strings and objects in message (while still keeping the object JSON parse-able)
2. OR, Is there any way I can remove the property from CustomDimensions
3. OR, Is there any other field that I can use to pass the message so that it gets in my message layout, but not as an event-property
4. OR, Can I achieve Json message in Raw without using JsonLayout, or by developing a custom layout (I couldn't see how that'll work)

**Comment by snakefoot** [comment]
Have you tried to specify `format=@`:

```xml
<targets>
    <target name="logfile" xsi:type="File" layout="${event-properties:item=myObject}${message}" />
    <target xsi:type="ApplicationInsightsTarget" name="aiTarget">
      <layout xsi:type="JsonLayout">
        <attribute name="MyItem" layout="${mdlc:item=MyItem:whenEmpty=Default}"/>
        <attribute name="Message" layout="${message}"/>
        <attribute name="myObject" layout="${event-properties:item=myObject:format=@}" encode='false' />
      </layout>
    </target>
</targets>
```

**Comment by pro-panda** [comment]
No, it didn't work

**Comment by snakefoot** [comment]
> No, it didn't work

It is not easy to help, when you don't give any hints of what happened. Can you provide a small example-project where fails?
